examples/Tag_caption_images_with_GPT4V.ipynb

```diff
@@ -0,0 +1,2546 @@
+      "--------------------------\n",
+      "\n",
+      "Pickleball Doormat, Welcome Doormat Absorbent Non-... - https://www.amazon.com/dp/B0C1MRB2M8 :\n",
+      "\n",
+      "This is a rectangular doormat featuring a playful design that caters to pickleball enthusiasts. The mat's background is a natural coir color, which is a fibrous material made from coconut husks, known for its durability and excellent scraping properties. Emblazoned across the mat in bold, black letters is the phrase \"it's a good day to play PICKLEBALL,\" with the word \"PICKLEBALL\" being prominently displayed in larger font size. Below the text, there are two crossed pickleball paddles in black, symbolizing the sport.\n",
+      "\n",
+      "The doormat measures approximately 16x24 inches, making it a suitable size for a variety of entryways. Its design suggests that it has an absorbent quality, which would be useful for wiping shoes and preventing dirt from entering the home. Additionally, the description implies that the doormat has a non-slip feature, which is likely due to a backing material that helps keep the mat in place on various floor surfaces. This mat would be a welcoming addition to the home of any pickleball player or sports enthusiast, offering both functionality and a touch of personal interest.\n",
+      "--------------------------\n",
+      "\n",
+      "JOIN IRON Foldable TV Trays for Eating Set of 4 wi... - https://www.amazon.com/dp/B0CG1N9QRC :\n",
+      "\n",
+      "This image features a set of four foldable TV trays with a stand, designed for eating or as snack tables. The tables are presented in a sleek grey finish, which gives them a modern and versatile look, suitable for a variety of home decor styles. Each tray table has a rectangular top with a wood grain texture, supported by a sturdy black iron frame that folds easily for compact storage. The accompanying stand allows for neat organization and easy access when the tables are not in use. These tables are ideal for small spaces where multifunctional furniture is essential, offering a convenient surface for meals, work, or leisure activities.\n",
+      "--------------------------\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "for index, row in examples.iterrows():\n",
+    "    print(f\"{row['title'][:50]}{'...' if len(row['title']) > 50 else ''} - {row['url']} :\\n\")\n",
+    "    img_description = describe_image(row['primary_image'], row['title'])\n",
+    "    print(f\"{img_description}\\n--------------------------\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d63f628f",
+   "metadata": {},
+   "source": [
+    "### Turning descriptions into captions\n",
```

Comment:
Did you try going straight to captions instead of via descriptions and it didn't work? if it didn't work well then this makes sense otherwise might be better to try straight to captions

Reply:
That was to give few shot examples without having to provide an image and as I suggest fine-tuning if few-shot examples aren't enough I thought it would make more sense to keep it separate!